GitCommands/CustomDiffMergeToolCache.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using GitCommands.Config;
+using GitUI;
+
+namespace GitCommands
+{
+    public class CustomDiffMergeToolCache
+    {
+        private CustomDiffMergeToolCache(bool isDiff)
+        {
+            _isDiff = isDiff;
+        }
+
+        private bool _isDiff { get; set; }
+        private SemaphoreSlim _mutex = new(1);
+        private IEnumerable<string>? _tools = null;
+
+        public static CustomDiffMergeToolCache DiffToolCache { get; } = new(true);
+        public static CustomDiffMergeToolCache MergeToolCache { get; } = new(false);
+
+        /// <summary>
+        /// Clear the existing caches
+        /// </summary>
+        public void Clear()
+        {
+            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await _mutex.WaitAsync().ConfigureAwait(false);
+
+                try
+                {
+                    _tools = null;
+                }
+                finally
+                {
+                    _mutex.Release();
+                }
+            }).FileAndForget();
+        }
+
+        /// <summary>
+        /// Load the availble DiffMerge tools and apply to the menus
+        /// </summary>
+        /// <param name="module">The Git module</param>
+        /// <param name="delay">The delay before starting the operation</param>
+        public async Task<IEnumerable<string>> GetToolsAsync(GitModule module, int delay)
+        {
+            if (_tools != null)
```

Comment:
[nit]
```suggestion
            if (_tools is not null)
```

Few other instances throughout

Reply:
old habits die slow